### PR TITLE
fix: fix parsing of IPv6 addresses in coder port-forward

### LINKED ISSUE #15561: coder port-forward with IPv6 local address fails parsing
`coder port-forward` supports setting the local address like:

```
coder port-forward <workspace> --tcp 1.2.3.4:8080:8080
```

However, our parsing of this is broken for IPv6 addresses, which should be entered like `[::1]:8080:8080`. This means it's not possible to get a local IPv6 listener on localhost.

Relates to https://github.com/coder/coder/issues/14529

### PR DESCRIPTION
**Pull request by spikecurtis**
fixes: #15561

Fixes parsing of IPv6 local addresses on `coder port-forward`

### CODE REVIEW

**Comment by spikecurtis** [comment]
* **#15627** <a href="https://app.graphite.dev/github/pr/coder/coder/15627?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/coder/coder/15627?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by spikecurtis** [comment]
FYI, I'm planning to follow up with a PR that defaults to opportunistically binding to IPv4 and IPv6 if you don't explicitly set the address.

File: cli/portforward.go [hunk 1]
```diff
@@ -326,63 +327,51 @@ type parsedSrcDestPort struct {
...
 }
 
+// specRegexp matches port specs. It handles all the following formats:
+//
+// 8000
+// 8888:9999
+// 1-5:6-10
+// 8000-8005
+// 127.0.0.1:4000:4000
+// [::1]:8080:8081
```
Comment (deansheather) [inline cli/portforward.go:337]: Should list ranges for specs with IPs too right?

**Review by deansheather — APPROVED** [review]

**Review by ethanndickson — APPROVED** [review]

**Comment by spikecurtis** [comment]
### Merge activity

* **Nov 25, 5:07 AM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/coder/coder/15627).

**Pull request merged**